### PR TITLE
MAYA-115180 - Copy MaterialXGenOgsXml to MayaUSD

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
Copied from [adsk_contrib/dev branch](https://github.com/autodesk-forks/MaterialX/tree/adsk_contrib/dev/source/MaterialXGenOgsXml) of Autodesk form of MaterialX

Allows MayaUSD to use a regular ILM MaterialX release. Will work with upcoming
v1.38.3.

### CODE REVIEW

File: lib/mayaUsd/render/MaterialXGenOgsXml/OgsXmlGenerator.cpp [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+MATERIALX_NAMESPACE_BEGIN
+
+namespace {
+typedef std::unordered_map<string, const pugi::char_t*> OGS_TYPE_MAP_T;
+
+const OGS_TYPE_MAP_T& getOgsTypeMap()
+{
+    // Data types used by OGS
+    // Delayed initialization to survive C++ init order fiasco.
+    // Keyed by string to survive multiple redinitions of global symbols.
```
Comment (williamkrick) [inline lib/mayaUsd/render/MaterialXGenOgsXml/OgsXmlGenerator.cpp:26]: Type redinitions

**Review by williamkrick — DISMISSED** [review]
It looks reasonable to me but I am not a materialX expert so I can't verify if the materialX usage is correct or not.

**Comment by seando-adsk** [comment]
@JGamache-autodesk I removed the ready for merge label. You need to get a code review completed before I can merge it. Also can you look at the security/flawfinder ?

**Comment by JGamache-autodesk** [comment]
Security/flawfinder is all about the added PugiXML library. Not much I can do here.
Mitigations:
1 - I upgraded to the latest version which has more static analysis done
2 - Used only for writing, so all those "If reading malicious input" warnings do not apply.

**Comment by JGamache-autodesk** [comment]
@kwokcb Do you see any issue with this MaterialXGenOgsXml integration into MayaUSD?

**Review by williamkrick — APPROVED** [review]

**Pull request merged**